corehq/apps/userreports/pillow.py

```diff
@@ -711,3 +711,32 @@ def get_location_pillow(pillow_id='location-ucr-pillow', include_ucrs=None,
         change_processed_event_handler=event_handler,
         processor=[ucr_processor]
     )
+
+
+def get_kafka_ucr_registry_pillow(
+    pillow_id='kafka-ucr-registry',
+    num_processes=1, process_num=0, dedicated_migration_process=False,
+    processor_chunk_size=DEFAULT_PROCESSOR_CHUNK_SIZE, ucr_configs=None, **kwargs):
+    """UCR pillow that reads from all 'case' Kafka topics and writes data into the UCR database tables
+
+    Only UCRs backed by Data Registries are processed in this pillow.
+
+        Processors:
+          - :py:class:`corehq.apps.userreports.pillow.ConfigurableReportPillowProcessor`
+    """
+    table_manager = RegistryDataSourceTableManager(
+        run_migrations=(process_num == 0)  # only first process runs migrations
+    )
+    ucr_processor = ConfigurableReportPillowProcessor(table_manager)
+    if ucr_configs:
+        table_manager.bootstrap(ucr_configs)
```

Comment:
Can you explain this part to me?  When will `ucr_configs` be populated vs empty?  What are the implications of running a pillow with an unbootstrapped table manager?

Reply:
This is mostly used for tests. The list is empty / None then the processor will bootstrap itself on the first change.